### DISCUSSION TITLE
Installing from source: missing "make uninstall"

### DISCUSSION THREAD

**Discussion by o-alquimista** [transmission/transmission#6354] [Q&A] [answered]
There is no target for `make uninstall`. Is it okay to overwrite an existing installation on upgrades?

I took note of the install log this time, and I plan to use it to manually uninstall:
```
-- Install configuration: "RelWithDebInfo"
-- Up-to-date: /usr/local/share/transmission/public_html
-- Installing: /usr/local/share/transmission/public_html/transmission-app.css
-- Installing: /usr/local/share/transmission/public_html/transmission-app.js.LEGAL.txt
-- Installing: /usr/local/share/transmission/public_html/index.html
-- Up-to-date: /usr/local/share/transmission/public_html/images
-- Installing: /usr/local/share/transmission/public_html/images/favicon.ico
-- Installing: /usr/local/share/transmission/public_html/images/favicon.png
-- Installing: /usr/local/share/transmission/public_html/images/webclip-icon.png
-- Installing: /usr/local/share/transmission/public_html/transmission-app.js
-- Installing: /usr/local/share/doc/transmission/AUTHORS
-- Installing: /usr/local/share/doc/transmission/COPYING
-- Installing: /usr/local/share/doc/transmission/README.md
-- Installing: /usr/local/share/doc/transmission/rpc-spec.md
-- Installing: /usr/local/share/doc/transmission/send-email-when-torrent-done.sh
-- Up-to-date: /usr/local/share/doc/transmission/news
-- Installing: /usr/local/share/doc/transmission/news/news-2.84.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.77.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.94.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.51.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.92.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.76.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.91.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.52.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.71.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.81.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.4.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.3.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.0-beta-2.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.82.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.1.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.5.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.73.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.80.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.93.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.50.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.83.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.0.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.0-beta-1.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.2.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.42-and-older.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.75.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.60.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.72.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.74.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.61.md
-- Installing: /usr/local/share/doc/transmission/news/news-3.00.md
-- Installing: /usr/local/share/doc/transmission/news/news-4.0.0-beta-3.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.70.md
-- Installing: /usr/local/share/doc/transmission/news/news-2.90.md
-- Installing: /usr/local/bin/transmission-daemon
-- Installing: /usr/local/share/man/man1/transmission-daemon.1
-- Installing: /usr/local/bin/transmission-create
-- Installing: /usr/local/share/man/man1/transmission-create.1
-- Installing: /usr/local/bin/transmission-edit
-- Installing: /usr/local/share/man/man1/transmission-edit.1
-- Installing: /usr/local/bin/transmission-remote
-- Installing: /usr/local/share/man/man1/transmission-remote.1
-- Installing: /usr/local/bin/transmission-show
-- Installing: /usr/local/share/man/man1/transmission-show.1
```

I'm tracking the `4.0.x` branch. Distribution packages are too old. Might try flatpak, because it takes a long time to compile on the Raspberry Pi.

**Comment by ckerr** [comment, marked as answer]
https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#can-i-do-make-uninstall-with-cmake discusses the pros & cons of a `make uninstall` for cmake.

tldr for Unix users, `$ xargs rm < install_manifest.txt`
